### ISSUE TITLE
intermittent error:   jwks_uri must be configured

### ISSUE THREAD

**Issue by llamahunter** [keattang/eks-auth-proxy#9]
I am using and Okta OIDC provider, and using this to access the kubernetes-dashboard.  Sometimes, for a while, the eks-auth-proxy emits this error to the log and I am unable to
retrieve a token for the dashboard to log in.

```
2020-05-09T19:57:54: PM2 log: Launching in no daemon mode
2020-05-09T19:57:54: PM2 log: App [index:0] starting in -fork mode-
2020-05-09T19:57:54: PM2 log: App [index:0] online
info:    Proxying requests to http://kubernetes-dashboard
info:    Proxy server running on port 3001
error:   jwks_uri must be configured
```

If I kill the eks-auth-proxy pod a few times and let it restart, it will usually start working. Very little info seems available about what this error message means.  I was able to find only _one_ hit in Google (a near googlewhack?)
https://npmdoc.github.io/node-npmdoc-openid-client/build/apidoc.html

I assume okta is doing something bad, but then why does it fix itself after restarting the eks-auth-proxy a few times?

**Comment by jin-ahn** [comment]
I am also getting this issue. The frustrating thing is that once it happens, we have to go in manually restart the container or launch a new one. Makes this unusable.

**Comment by keattang** [comment]
This is an odd issue. The way things are set up, it should get the `jwks_uri` from the OIDC discovery URL configured with --oidc-issuer` so it appears that Okta is not always returning that property. 

It seems someone else is seeing this issue and found that using a different discovery URL might help: https://devforum.okta.com/t/no-jwks-uri-in-well-known-oauth-authorization-server-response/9876

I don't have an Okta account so would someone be able to try this out and report back?

**Comment by llamahunter** [comment]
can confirm that jwks_uri does not appear in the `/.well-known/oauth-authorization-server` endpoint, but does appear in the `/.well-known/openid-configuration` endpoint.

```
% curl -s https://xperi.okta.com//.well-known/oauth-authorization-server | jq .jwks_uri
null
```

```
% curl -s https://xperi.okta.com//.well-known/openid-configuration | jq .jwks_uri
"https://xperi.okta.com/oauth2/v1/keys"
```

**Comment by llamahunter** [comment]
Possibly eks-auth-proxy is using an old version of the openid-client?  See this change in 5.0.0 that deprecates calling the oauth-authorization-server endpoint.
https://github.com/panva/node-openid-client/blob/296e25d877f7f2a338978a6d5ce3c94b28548646/CHANGELOG.md#-breaking-changes

Note that currently eks-auth-proxy appears pinned to version openid-client 4.4.0

**Comment by llamahunter** [comment]
any chance for an update to the openid-client dependency?

**Comment by keattang** [comment]
Yep. Currently working on upgrading all dependencies but getting some odd errors that I need to work through. Hoping to have something out in the next week or two.

**Comment by keattang** [comment]
@llamahunter this should be fixed now in 2.0.0
